File: Competitive Programming/cnote.java
import java.util.*;
import java.io.*;
public class cnote
{
    public static void main(String args[])throws IOException
    {
        int t,n,k,x,y;
        InputStreamReader read=new InputStreamReader(System.in);
        BufferedReader br=new BufferedReader(read);
        t=Integer.parseInt(br.readLine());
        while(t-- !=0)
        {
            x=Integer.parseInt(br.readLine());
            y=Integer.parseInt(br.readLine());
            k=Integer.parseInt(br.readLine());
            n=Integer.parseInt(br.readLine());
            int ar[]=new int[n];
            int arr[]=new int[n];
            for(int i=0;i<n;i++)
            {
                 ar[i]=Integer.parseInt(br.readLine());
                 arr[i]=Integer.parseInt(br.readLine());
            }

            int c=x-y;int i;
            for(i=0;i<n;i++)
            {
               if(ar[i]>=c && arr[i]<=k)
               {
                   System.out.println("LuckyChef");
                   break;
               }

            }
            if(i==n)
                System.out.println("UnluckyChef");

        }
    };
};

File: Competitive Programming/cnote.c
#include<stdio.h>
int main()
{
    long long int t,x,y,k,n,p,c,i;
    scanf("%ld",&t);
    while(t--)
    {
        scanf("%lld %lld %lld %lld",&x,&y,&k,&n);
        y=x-y;
        int flag=0;
        for(i=0;i<n;i++)
        {
            scanf("%lld",&p);
            scanf("%lld",&c);
            if(p>=y && c<=k)
                flag=1;
        }

         if(flag==0)
            printf("UnluckyChef\n");
         else
            printf("LuckyChef\n");
    }
    return 0;
}
